### PR TITLE
✨ Add support for CRD singular name

### PR DESCRIPTION
**Pull request by damemi**
This adds a `+kubebuilder:singular=foo` tag to specify singular names for CRDs if desired

### CODE REVIEW

**Comment by damemi** [comment]
On an unrelated note, the PR submission says to see VERSIONING.md for which icon to add, but I couldn't find any versioning file in this repo. So I have gone with sparkles based on looking at a few other PRs, if this is incorrect I can update.

**Comment by DirectXMan12** [comment]
whoops.  [VERSIONING.md is here](https://github.com/kubernetes-sigs/controller-runtime/blob/master/VERSIONING.md) (in controller-runtime).

File: pkg/internal/codegen/parse/util.go [hunk 1]
```diff
@@ -311,6 +325,16 @@ func getCategoriesTag(c *types.Type) string {
 	return resource
 }
 
+// getSingularName returns the value of the +kubebuilder:singular tag
+func getSingularName(c *types.Type) string {
+	comments := Comments(c.CommentLines)
+	singular := comments.getTag("kubebuilder:singular", "=")
+	if len(singular) == 0 {
+		panic(errors.Errorf("Must specify +kubebuilder:singular comment for type %v", c.Name))
```
Comment (DirectXMan12) [inline pkg/internal/codegen/parse/util.go]: this error doesn't make any sense with how singular is used.  maybe reword to say "must specify a value to use with singular" or something?

Comment (sohankunkerkar) [reply pkg/internal/codegen/parse/util.go]: @damemi  could you please add one test case here for this change? You can add your test case here https://github.com/kubernetes-sigs/controller-tools/blob/master/pkg/crd/generator/testData/pkg/apis/fun/v1alpha1/toy_types.go#L97
and run this file to see if you're getting the proper o/p.
https://github.com/kubernetes-sigs/controller-tools/blob/master/pkg/crd/generator/generator_test.go

**Comment by damemi** [comment]
@DirectXMan12 @sohankunkerkar thanks, I've updated the error message and added a test case for this. Could either of you please take another look?
